src/GraphQL.StarWars/StarWarsQuery.cs

```diff
@@ -9,16 +9,16 @@ public StarWarsQuery(StarWarsData data)
     {
         Name = "Query";
 
-        Field<CharacterInterface>("hero", resolve: context => data.GetDroidByIdAsync("3"));
-        Field<HumanType>(
+        FieldAsync<CharacterInterface>("hero", resolve: async context => await data.GetDroidByIdAsync("3"));
```

Comment:
I think we should consider unify all `Field` overloads before v5. I would prefer use `Field` instead of `FieldAsync` but you chose `ResolveAsync` instead of `Resolve` for resolvers. In that way `FieldAsync` seems more logical though.

Reply:
I'm open to suggestions.  Here's my thoughts so far and we can discuss:
1. All methods that return `Task` or `ValueTask` should end in `Async`.  `ResolveAsync` returns a `ValueTask<object>`, so it must be `ResolveAsync` (or `SubscribeAsync`).
2. `Field` isn't an asynchronous method, so it would be appropriate to name them all `Field`.  But we need to consider (a) backwards compatibility, (b) we may need a way to differentiate the method overload for synchronous and asynchronous methods.
3. Are we supporting both `Task` asynchronous methods and `ValueTask` asynchronous methods?  If only one, which one?

I think we need to create a '[Discussion]' issue and discuss this further before a change is made.

Reply:
Also we need to remember that the generic type for `ValueTask` and `Task` are not covariant.  (not `out T`)  This can be a pain when assigning methods to field resolvers.

Reply:
OK